src/data/repositories/SurveyFormD2Repository.ts

```diff
@@ -85,14 +89,25 @@ export class SurveyD2Repository implements SurveyRepository {
         return apiToFuture(
             this.api.request<ProgramMetadata>({
                 method: "get",
-                url: `/programs/${programId}/metadata.json?fields=programs,dataElements,programStageDataElements,programStageSections,trackedEntityAttributes,programStages`,
+                url: `/programs/${programId}/metadata.json?fields=programs,dataElements,programStageDataElements,programStageSections,trackedEntityAttributes,programStages,programRules,programRuleVariables,programRuleActions`,
             })
         ).flatMap(resp => {
             if (resp.programs[0]) {
                 const programDataElements = resp.programStageDataElements.map(
                     psde => psde.dataElement
                 );
 
+                const dataElementsWithSortOrder: ProgramDataElement[] = resp.dataElements.map(
+                    de => {
+                        return {
+                            ...de,
+                            sortOrder: resp.programStageDataElements.find(
+                                psde => psde.dataElement.id === de.id
+                            )?.sortOrder,
+                        };
+                    }
+                );
+
                 //If event specified,populate the form
                 if (eventId) {
```

Comment:
Not from this PR but maybe it's better in these ifs to create functions with the code and naming the functions to describe what is being done in each one

Reply:
the conditions do have functions associated to each of them, however the length of each condition is due to large number of params. I will address this in the next PR, need some time to evaluate collapsing the params vs creating seperate functions.